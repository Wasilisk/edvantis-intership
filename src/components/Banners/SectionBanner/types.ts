export type SectionBannerStylesProps = {
    backgroundImage: string
}