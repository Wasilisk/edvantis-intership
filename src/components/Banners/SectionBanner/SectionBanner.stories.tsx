import {ComponentMeta, Story} from "@storybook/react";
import React from "react";
import {SectionBannerType} from "../../../models";
import SectionBanner from "./index";

const mockSectionBanner: SectionBannerType = {
    banner_texture: {
        url: "https://res2.weblium.site/res/5e2f1da2c947890021c3f194/5e341fc418c4d700210727e5"
    },
    title: {
        "highlight": "Top Collections",
        "after_highlight": "of Designer Cushioned Furniture"
    }
}

export default {
    title: 'Application/Banners/SectionBanner',
    component: SectionBanner,
} as ComponentMeta<typeof SectionBanner>;

const Template: Story<SectionBannerType> = (props) => <SectionBanner {...props}/>;

export const Default = Template.bind({});
Default.args = {
    ...mockSectionBanner
};
