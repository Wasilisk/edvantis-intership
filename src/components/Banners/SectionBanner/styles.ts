import styled from "styled-components";
import {SectionBannerStylesProps} from "./types";

export const SectionBannerStyles = styled.div<SectionBannerStylesProps>`
  width: 100%;
  padding: 160px 0;
  display: flex;
  justify-content: center;
  background-image: ${({backgroundImage}) => `url(${backgroundImage});`};
  background-position:  center center / 22px;
  background-repeat: repeat;

  .banner-info {
    max-width: 1200px;
    text-align: center;
    padding: 0 15px;

    & > h2 {
      font-size: 84px;
      margin: 0;
      overflow-wrap: anywhere;
    }
  }


  @media (max-width: 1024px) {
    .banner-info {
      h2 {
        font-size: 70px;
      }
    }
  }

  @media (max-width: 768px) {
    padding: 130px 0;
    .banner-info {
      h2 {
        font-size: 56px;
      }
    }
  }
  
  @media (max-width: 320px) {
    padding: 100px 0;
    .banner-info {
      h2 {
        font-size: 40px;
      }
    }
  }
`