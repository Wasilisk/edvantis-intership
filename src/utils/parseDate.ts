const month = ["January","February","March","April","May","June","July","August","September","October","November","December"];

export const parseDate = (timestamp: number) => {
    const date = new Date(timestamp);

    return `${month[date.getMonth()]} ${('0' + date.getDate()).slice(-2)}, ${date.getFullYear()}`;
}