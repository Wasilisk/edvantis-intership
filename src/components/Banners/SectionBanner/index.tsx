import React from 'react';
import {SectionBannerStyles} from './styles';
import {useAppSelector} from "../../../hooks";
import {selectSectionBanner} from "../../../store/slices/HomePage/selectors";

const SectionBanner = () => {
    const sectionBannerContent = useAppSelector(selectSectionBanner);

    return (
        sectionBannerContent && <SectionBannerStyles backgroundImage={sectionBannerContent.banner_texture.url}>
            <div className="banner-info">
                <h2>
                    {
                        sectionBannerContent.title.highlight &&
                        <span className="highlight">{sectionBannerContent.title.highlight}&nbsp;</span>
                    }
                    {sectionBannerContent.title.plain_text}
                </h2>
            </div>
        </SectionBannerStyles>
    );
};

export default SectionBanner;