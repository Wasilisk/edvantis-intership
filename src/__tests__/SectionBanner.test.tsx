import {render} from "./utils/store-wrapper";
import {mockSectionBanner} from "./__mock__";
import SectionBanner from "../components/Banners/SectionBanner";

describe('SectionBanner component', () => {
    it('should render component', () => {
        const {getByText, container, asFragment} = render(<SectionBanner {...mockSectionBanner}/>);

        expect(container.firstChild).toHaveStyle(`background-image: url(${mockSectionBanner.banner_texture.url})`);

        const highlightText = getByText(mockSectionBanner.title.highlight!);
        expect(highlightText).toBeDefined();

        const plainText = getByText(mockSectionBanner.title.after_highlight!);
        expect(plainText).toBeDefined();

        expect(asFragment()).toMatchSnapshot();
    })
})