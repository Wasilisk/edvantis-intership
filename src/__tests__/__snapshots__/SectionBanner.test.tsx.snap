// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section Banner component should render component 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ jWCUBa"
  >
    <div
      class="banner-info"
    >
      <h2>
        <span
          class="highlight"
        >
          Styled textÂ 
        </span>
        Simple text
      </h2>
    </div>
  </div>
</DocumentFragment>
`;

exports[`SectionBanner component should render component 1`] = `
<DocumentFragment>
  <div
    class="sc-bczRLJ jWCUBa"
  >
    <div
      class="banner-info"
    >
      <h2>
        <span
          class="highlight"
        >
          Styled text 
        </span>
        Simple text
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
